app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala

```diff
@@ -46,6 +49,16 @@ class AuthenticatedActionBuilder @Inject()(
           case (None, Some(_)) => Future.successful(Unauthorized(Json.obj("errorCode" -> "NO_EXTERNAL_ID")))
           case (None, None) => Future.successful(Unauthorized(Json.obj("errorCode" -> "NO_EXTERNAL_ID_OR_GROUP_ID")))
         }
+    }recoverWith {
+      case e: InternalError => {
```

Comment:
if you are throwing all at the end you dont need this.

Reply:
Internal error is a child of AuthorisationException

Reply:
ok, that one is fine then.